src/main/java/org/jfrog/hudson/pipeline/common/executors/ConanExecutor.java

```diff
@@ -0,0 +1,196 @@
+    }
+
+    private void persistBuildProperties(FilePath conanHomeDirectory) throws IOException, InterruptedException {
+        FilePath buildProperties = new FilePath(conanHomeDirectory, ".conan").child("artifacts.properties");
+        ;
+        final long startTime = buildInfo.getStartDate().getTime();
+        final Vcs vcs = Utils.extractVcs(ws, new JenkinsBuildInfoLog(listener));
+        buildProperties.touch(System.currentTimeMillis());
+        buildProperties.act(new PersistBuildPropertiesCallable(buildInfo, startTime, vcs));
+    }
+
+    public static class PersistBuildPropertiesCallable extends MasterToSlaveFileCallable<Boolean> {
+        private static final String PROPS_PREFIX = "artifact_property_";
+        private BuildInfo buildInfo;
+        private long startTime;
+        private Vcs vcs;
+
+        PersistBuildPropertiesCallable(BuildInfo buildInfo, long startTime, Vcs vcs) {
+            this.buildInfo = buildInfo;
+            this.startTime = startTime;
+            this.vcs = vcs;
+        }
+
+        public Boolean invoke(File conanProperties, VirtualChannel channel) throws IOException, InterruptedException {
+            try (PrintWriter pw = new PrintWriter(new FileWriter(conanProperties.getCanonicalFile()));) {
+                pw.println(String.format("%s=%s", PROPS_PREFIX + BuildInfoFields.BUILD_NAME, buildInfo.getName()));
+                pw.println(String.format("%s=%s", PROPS_PREFIX + BuildInfoFields.BUILD_NUMBER, buildInfo.getNumber()));
+                pw.println(String.format("%s=%s", PROPS_PREFIX + BuildInfoFields.BUILD_TIMESTAMP, String.valueOf(startTime)));
+                if (StringUtils.isNotEmpty(vcs.getRevision())) {
+                    pw.println(String.format("%s=%s", PROPS_PREFIX + BuildInfoFields.VCS_REVISION, vcs.getRevision()));
+                }
+                if (StringUtils.isNotEmpty(vcs.getUrl())) {
+                    pw.println(String.format("%s=%s", PROPS_PREFIX + BuildInfoFields.VCS_URL, vcs.getUrl()));
+                }
+            }
+            return true;
+        }
+    }
+
+    // Conan collect buildInfo as part of the tasks execution.
+    // In order to transform the collected buildInfo into Artifactory buildInfo format we need to execute the conan_build_info command.
+    // The conan_build_info command expect to get a path for the output file.
+    private FilePath collectConanBuildInfo(EnvVars extendedEnv) throws Exception {
+        FilePath tempDir = ExtractorUtils.createAndGetTempDir(ws);
+        FilePath outputFilePath = tempDir.createTextTempFile("conan", "build-info", "", true);
+        // We will use this executor to run conan's collect build info command
+        conanCmdArgs = new ArgumentListBuilder();
+        conanCmdArgs.add(CONAN_BUILD_INFO_CMD);
+        conanCmdArgs.add(getLogFilePath());
+        conanCmdArgs.add("--output");
+        conanCmdArgs.add(outputFilePath.getRemote());
```

Comment:
I thinks it can be shorted to this:
```suggestion
        conanCmdArgs = new ArgumentListBuilder(CONAN_BUILD_INFO_CMD, getLogFilePath(), "--output", outputFilePath.getRemote());
```